From de4900c9d64c0ca0311afc83469bfc379e3bab52 Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Mon, 11 May 2020 11:49:44 +0200
Subject: [PATCH] linux-qoriq: arm64 kernel now conserves some power

When the kernel stops/crashes cpu_do_idle() is called

Signed-off-by: Heinz Wrobel <Heinz.Wrobel@nxp.com>
---
 arch/arm64/kernel/process.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/kernel/process.c b/arch/arm64/kernel/process.c
index d668c13218b8..053957ab11ae 100644
--- a/arch/arm64/kernel/process.c
+++ b/arch/arm64/kernel/process.c
@@ -118,7 +118,8 @@ void machine_halt(void)
 {
 	local_irq_disable();
 	smp_send_stop();
-	while (1);
+	while (1)
+		cpu_do_idle();
 }
 
 /*
@@ -167,7 +168,8 @@ void machine_restart(char *cmd)
 	 * Whoops - the architecture was unable to reboot.
 	 */
 	printk("Reboot failed -- System halted\n");
-	while (1);
+	while (1)
+		cpu_do_idle();
 }
 
 static void print_pstate(struct pt_regs *regs)
-- 
2.17.1

